kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/config/CredentialsValidatorTest.java

```diff
@@ -66,4 +67,28 @@ public void testFailureToConstructClient() {
         new CredentialsValidator.GcsCredentialsValidator().doValidate(config)
     );
   }
+
+  @Test
+  public void testApplicationDefaultCredentials() {
+    BigQuerySinkConfig config = mock(BigQuerySinkConfig.class);
+    when(config.getKey()).thenReturn(null);
+    when(config.getKeySource()).thenReturn(GcpClientBuilder.KeySource.APPLICATION_DEFAULT);
+
+    assertNotEquals(
+            Optional.empty(),
```

Comment:
This looks like we're testing what should be a green code path (validating our ability to construct a client with application default credentials), but the test here is asserting that the validation fails. Is this intentional?

Reply:
Yes, it is. Actually we were not previously testing the ability to construct clients (neither `testNoCredentialsSkipsValidation` nor `testFailureToConstructClient`) with the existing key sources. I don't even know if this makes sense at unit testing - we can not perform actual constructions against GCP. We could mock it, but then the test would be worthless, I would say. 

So, the added test testApplicationDefaultCredentials is just useful to test that the execution gets to the construction point. Thoughts?

Reply:
Good point RE: the existing tests. I think it's fine to leave this test case out as long as we have reasonable coverage with the newly-introduced `GcpClientBuilderIT` tests, which appears to be the case.